### ISSUE TITLE
[Waiting for payment] [$250] Editing date of an IOU details marks the main report as unread even though there is no new message in the main report

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#25949]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open an existing chat with user B
2. Click on '+' icon > click on 'Request money'
3. Enter amount & click on 'Next' button
4. Click on 'Request' button
5. Open a tab where user B is signed in
6. Open the main report or chat with user B, make sure that it is marked as read, and open chat with Concierge
7. Open a tab where user A is signed in
8. Click on IOU preview component
9. Click on IOU preview component of IOU thread report
10. Click on 'Date' field
11. Click on year dropdown menu
12. Select another date that is in the past & click on 'Save' button
13. Open a tab where user B is signed in

## Expected Result:
 The main report with user A shouldn't be marked as unread since there is no new message in the report

## Actual Result:
The main report with user A is marked as read

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.57-3

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):** 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/229f6865-a5f8-4c46-b9e7-46bc2e10d671

https://github.com/Expensify/App/assets/115492554/b2ec326d-0011-4d31-872c-ff1032c13a35

**Expensify/Expensify Issue URL:**

**Issue reported by:** @Natnael-Guchima

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692034383213899

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01163f580e40dfceb5</li>
        <li>Upwork Job ID: 1696674689918205952</li>
        <li>Last Price Increase: 2023-12-04</li>
    </ul>
</details>

**Comment by strepanier03** [comment]
@dangrous - Would you please attempt to repro this on DEV for me? I do not have a DEV environment to test.

Thank you for your help!

**Comment by Natnael-Guchima** [comment]
This is now reproducible on staging too  @strepanier03.

**Comment by dangrous** [comment]
@strepanier03 let me know if you can reproduce on staging - if not I can try on dev!

**Comment by strepanier03** [comment]
@dangrous - I am unable to repro this on staging. @Natnael-Guchima - What device and browser/app are you able to repro on staging. 

I attempted on Mac/Chrome and the chat doesn't become unread for me.

**Comment by dangrous** [comment]
Okay confirmed that this happens on Dev! 

I'm not 100% convinced it's a bug since the request DID change, but it didn't change visibly in the parent report so probably worth looking into

**Comment by strepanier03** [comment]
Thank you @dangrous - I will move this forward.

**Comment by sofi-a** [comment]
@Natnael-Guchima @dangrous @strepanier03 It looks like the last expense modification appears as a new message. Don't we consider expense modifications as report messages?

<img width="473" alt="Screenshot 2023-08-30 at 05 04 54" src="https://github.com/Expensify/App/assets/31987492/0b6fbc36-9e54-472e-a63c-94d8444d2712">

**Comment by sofi-a** [comment]
Contributor details
Your Expensify account email: sofoniasabathun@icloud.com
Upwork Profile Link: https://www.upwork.com/freelancers/~013c73af4810229810

**Comment by DylanDylann** [comment]
It seems like a BE Bug. The BE return new lastVisibleActionCreated for main report when editing date

**Comment by dangrous** [comment]
@sofi-a so that part **should** be unread; it's the parent report that's getting marked that (maybe) shouldn't - i.e. the original chat, not the request itself. Does that make sense? Let me know if I need to explain in a different way!

**Comment by sofi-a** [comment]
@dangrous Yes it does. Thank you. I'm still skeptical that this is still a bug though.

**Comment by dangrous** [comment]
Ha, I'm not confident either. I can see it both ways, either no bug - the request is updated so it should be marked as such (but maybe we should surface the request date in the preview?) or bug - the request does not visibly change so should not be marked as unread.

I do think that the user should definitely be notified that something has changed, but the unread marker in the actual request chat could be enough.

What do you think, @strepanier03 ? Should we bring this out in Slack somewhere to figure out what our ideal behavior/UX would be?

**Comment by dangrous** [comment]
Bump for the above when you have a moment @strepanier03 - do you think this a) is a bug, b) is not a bug, or c) should be brought up in Slack for more opinions?

**Comment by strepanier03** [comment]
Was out for a public holiday yesterday so just getting back to this.

> I do think that the user should definitely be notified that something has changed, but the unread marker in the actual request chat could be enough.

I agree with this @dangrous. I think the unread marker in the request chat should be enough , especially with the report message, so we shouldn't be marking the main report unread as that could be confusing with nothing posted in the main report to indicate why it's unread.

**Comment by dangrous** [comment]
Cool! However, now that I think about this more - I forgot that this was a **dev** bug only and not actually affecting real users. I think... we might be fine to close in that case, since the argument that it "could be confusing" doesn't really apply on dev. How does that sound?

It is a little peculiar that it only happens on dev, so that might be worth looking into at some point but doesn't seem very high priority

**Comment by Natnael-Guchima** [comment]
@dangrous last time I checked it was also reproducible on staging. I can check, and let you know if this is only reproducible on dev.

**Comment by dangrous** [comment]
got it - I tried on prod (could not reproduce) but not staging.

**Comment by Natnael-Guchima** [comment]
@dangrous  It seems to be reproducible on staging too. 

https://github.com/Expensify/App/assets/57330233/474de825-576d-4a34-acf9-0183d31d4456

**Comment by dangrous** [comment]
Yeah let's move forward and at least see if we can handle it externally. I think it might be internal but let's see what the community thinks!

**Comment by strepanier03** [comment]
Sounds good @dangrous, updating the title now and we'll see where it goes with the community involvement.

**Comment by dangrous** [comment]
I'll update the bounty here, to see if we can get more action!

**Comment by devrari** [comment]
I think  this is a  BE bug , lastReadTime from backend gets changed  to old time , because of which onyxmerge updates the lastReadTime and we see unread indicator.

**Comment by strepanier03** [comment]
Sounds like this needs to switch over to internal so I'll make that change now.

**Comment by dangrous** [comment]
Asking for discussion in [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1695008172130889) - I'm not really very sure what I think is the "best" solution to this, just from a product perspective, so hopefully hearing from others will help!

**Comment by strepanier03** [comment]
Thanks @dangrous, I'm following that thread and weighed in as well.

**Comment by dangrous** [comment]
Daily update, have x-posted in [#product](https://expensify.slack.com/archives/C03U7DCU4/p1695275966619789) to see if we can get some more eyes on this.

**Comment by strepanier03** [comment]
<img width="413" alt="image" src="https://github.com/Expensify/App/assets/10925636/2f69c3a3-4927-4415-a6a5-50933dbdbf90">

Things are moving forward. 

I'm OoO Mon-Tuesday next week so if there is a need for action from a BZ either day please post in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) to help.

**Comment by dangrous** [comment]
Haven't had time to get into this, hopefully later this week!

**Comment by strepanier03** [comment]
Thanks @dangrous, I'm back in office so ping me if you need to!

**Comment by mountiny** [comment]
Leaving a note here for myself:
- https://github.com/Expensify/Web-Expensify/blob/f5c62cffbf35107f75acd83403d3fd8566a50583/lib/ReportUtils.php#L1865

I think this is coming from here because we update the reportPreview action created time https://github.com/Expensify/Auth/blob/1ef827dd1e4db948a111790c47dc316069891824/auth/command/EditMoneyRequest.cpp#L332 but the lastReadTime of the report is probably unchanged

**Comment by mountiny** [comment]
no update

**Comment by mountiny** [comment]
Still no update, focusing on wave 6 and 7

**Comment by strepanier03** [comment]
Thanks @mountiny 🙌

**Comment by mountiny** [comment]
Still not able to prioritize this one

**Comment by dangrous** [comment]
I can grab this back from you if you like @mountiny though I also don't have much time to prioritize it haha

**Comment by mountiny** [comment]
Actually I would prefer i fyou could take it back, when I assigned myself it was just before getting some wave6 comittements and given I will be ooo some part of next week I need to look into those primarly

**Comment by mountiny** [comment]
Thank you!

**Comment by dangrous** [comment]
No worries! I'll dig in again probably on Monday

**Comment by dangrous** [comment]
Haven't had time yet, should do later this week

**Comment by dangrous** [comment]
Okay so I've realized this is a little weirder than I thought - but it might be intentional. I think there's at least something we can clean up, though. 

So the main thing - if a money request is edited, it not only gets marked as unread, it also *moves to the bottom of the conversation*. Which makes me think we did this on purpose. I just didn't notice it when testing this bug because I had never written any messages after the request. So, it sort of makes sense that it's treating it as a new message, if it literally moves it up. If that's how we want it, then I don't think we should change anything here. For reference, see the screenshots below, before and after updating the money request.

However:
1. Do we think this is a good idea? It feels like it would be very confusing. Should the request stay where it is, even if edited?
2. If we do decide to keep this behavior, the new chat line indicator seems to stay at the original location of the money request, at least on dev. That seems bad, and should be fixed.

@mountiny and @strepanier03 let me know what you think, and if we should maybe bring this up in Slack somewhere.

**BEFORE UPDATING REQUEST:**
<img width="479" alt="Screenshot 2023-10-24 at 14 11 48" src="https://github.com/Expensify/App/assets/5487802/cabdcb1a-a16b-4059-99e5-e352ac34a6d1">

**AFTER UPDATING REQUEST:**
<img width="489" alt="Screenshot 2023-10-24 at 14 12 02" src="https://github.com/Expensify/App/assets/5487802/a9da041d-1177-4da4-8967-24d0a69f4b76">

**Comment by mountiny** [comment]
I think we should not be updating the report preview timestamp since the modfiedexpense action is in the transaction thread and not directly in the expens report. However, I think it would be good to bring this up to slack to discuss and get broader aggreement

**Comment by dangrous** [comment]
posted here https://expensify.slack.com/archives/C03U7DCU4/p1698343200916719

**Comment by dangrous** [comment]
Getting a few more opinions it looks like; but again seems we're all agreed that we shouldn't update the timestamp (which should solve this issue)

**Comment by dangrous** [comment]
Everyone agrees! I will try to knock this out soonish

**Comment by dangrous** [comment]
Hookay so just removing [this code](https://github.com/Expensify/Auth/blob/1ef827dd1e4db948a111790c47dc316069891824/auth/command/EditMoneyRequest.cpp#L328-L333) does what we're hoping - preview still updates but not marked as unread or moved. But, the LHN preview still updates to `X owes $X`, until you click on the chat and then it returns to what it should be. So I need to figure out why that is... (@mountiny if you have any thoughts let me know). Otherwise pretty quick fix

**Comment by mountiny** [comment]
@dangrous what exactly does happen, do you have a video or numbered steps?

**Comment by dangrous** [comment]
Same steps as "Action Performed". Video:
https://github.com/Expensify/App/assets/5487802/75c69179-42b9-4e2e-88f1-168bcb6b7f95

**Comment by mountiny** [comment]
LHN preview of which chat, it should stay `X owes $X` for the expense report and for the workspace chat when you edit the request.

**Comment by strepanier03** [comment]
We're okay here Melvin. I'll keep following along until action is needed.

**Comment by dangrous** [comment]
> LHN preview of which chat, it should stay `X owes $X` for the expense report and for the workspace chat when you edit the request.

Oh, even the workspace chat should change to `X owes $X` when you edit the request? If so, I think it's the correct behavior, but @mountiny can you confirm from the video? it goes back to the latest chat once it's read. The expense report is working well.

**Comment by mountiny** [comment]
I see, this might be some pusher update we incorrectly sent when we edit the request I guess

**Comment by dangrous** [comment]
hm okay I can see what we send there. That's probably from Web-E, though, right? Not Auth? (I.e. I can push this change and go from there?

**Comment by mountiny** [comment]
I think these have not been migrated to Auth yet

**Comment by dangrous** [comment]
Okay cool. So how do we feel about:
1. Create a PR to update [this](https://github.com/Expensify/App/issues/25949#issuecomment-1787879097) in Auth to not mark the parent report as read
2. And only THEN creating a new PR in Web-E (Solution TBD) to fix the Pusher event not to update LHN.

Technically that might have the very slightly confusing behavior (in the video) for a little while until we figure out the solution to #2 but I think that's okay? Let me know if that sounds good to you.

**Comment by mountiny** [comment]
responded in the slack thread

**Comment by dangrous** [comment]
Responded in thread ;) https://expensify.slack.com/archives/C03U7DCU4/p1699574447577749?thread_ts=1698343200.916719&cid=C03U7DCU4

**Comment by dangrous** [comment]
Auth PR is merged, not yet on prod. Technically this bug is fixed once that's deployed but we should keep this open for the Pusher/Web-E fix. I'll look into that now but I might shift this issue to weekly since it's just a small part left (unless anyone objects?).

**Comment by dangrous** [comment]
Welp, can't quite figure this out yet. It's [this Onyx update](https://github.com/Expensify/Web-Expensify/blob/b6a77f014418fdf00dc6a0817520ad1ea2fdda79/lib/TransactionUtils.php#L1114-L1118) but I can't figure out which part exactly is sending the LHN update (vs. updating the money request). There's a chance this might be an App/frontend issue instead of Web-E?

**Comment by dangrous** [comment]
Okay [Web-E PR](https://github.com/Expensify/Web-Expensify/pull/39764) that should fix that LHN issue is up! Once that's deployed, i think we're good to close.

**Comment by strepanier03** [comment]
Got it, thanks @dangrous. I'll follow along to action once the PR is done.

**Comment by dangrous** [comment]
web-e is still on staging, should hopefully get to production soon!

**Comment by dangrous** [comment]
I think we're all set! @strepanier03 do you have a moment to re-test, and then handle any payment (this has gone on so long I'm not sure who needs what haha)

**Comment by strepanier03** [comment]
Sorry, this didn't have the normal naming convention for a GH awaiting payment so I missed it was ready. Reviewing this now.

**Comment by strepanier03** [comment]
Thank you @fedirjh - I appreciate you weighing in and for the reminder I didn't' realize this was so old but you're 100% right. 

I'll move forward with payment and finishing this up.

**Comment by strepanier03** [comment]
@Natnael-Guchima - I've hired you to the job, please accept when you can and I'll pay it out.

**Comment by Natnael-Guchima** [comment]
I have accepted the offer. Thanks @strepanier03

**Comment by strepanier03** [comment]
All right, this is paid out. Closing!
